### DISCUSSION TITLE
TFT to BITMAP (or other format)

### DISCUSSION THREAD

**Discussion by jaymedavis** [Bodmer/TFT_eSPI#3107] [Q&A - General]
Great library, thank you so much.  There are lots of examples going from an image to the display, but I was unable to find one going the other way - TFT display -> generate bitmap/img/png/etc.

I have a complex screen I have drawn out using this library as a sprite, and I'd like to generate a bitmap (or other format) of the LCD screen so I can send it to a web service.  Can you give me any advice on achieving this?  I have successfully used a loop for readPixel/drawPixel and drawn the same screen via that way, perhaps I could take that output (uint16_t zzxx[LCD_HEIGHT + 1] array) and generate an image somehow?  or perhaps there is a way to generate a graphic using the sprite?

Any advice appreciated!

**Comment by Bodmer** [comment]
Test the display works for reads from the CGRAM using the example sketch TFT_ReadWrite_Test. If that works then the screen can be read pixel by pixel with readPixel(x,y).

Pixel values can also be read from a sprite.

Have a look at the TFT_Screen_Capture example. It might not work unless readPixel is used but may give you some ideas.

I don't have any examples for creating standard image formats (png, jpg, bmp etc.
